velox/dwio/parquet/reader/ParquetReader.cpp

```diff
@@ -343,18 +344,62 @@ std::unique_ptr<ParquetTypeWithId> ReaderBase::getParquetColumnInfo(
           // also an indication of this is a list type
           // child of LIST
           VELOX_CHECK_GE(children.size(), 1);
-          auto type = TypeFactory<TypeKind::ARRAY>::create(children[0]->type());
-          return std::make_unique<ParquetTypeWithId>(
-              std::move(type),
-              std::move(children),
-              curSchemaIdx,
-              maxSchemaElementIdx,
-              ParquetTypeWithId::kNonLeaf, // columnIdx,
-              std::move(name),
-              std::nullopt,
-              std::nullopt,
-              maxRepeat,
-              maxDefine);
+          if (children.size() == 1 && name != "array" &&
+              !boost::algorithm::ends_with(name, "_tuple")) {
```

Comment:
actually the more accurate expression is `name is not parent.name() + "_tuple"` . I added a simplified version before for quick test. https://github.com/prestodb/presto/blob/master/presto-parquet/src/main/java/com/facebook/presto/parquet/ParquetTypeUtils.java#L95